exp/api/v1beta1/awsmachinepool_webhook.go

```diff
@@ -98,6 +98,16 @@ func (r *AWSMachinePool) validateSubnets() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSMachinePool) validateAdditionalSecurityGroups() field.ErrorList {
+	var allErrs field.ErrorList
+	for _, sg := range r.Spec.AWSLaunchTemplate.AdditionalSecurityGroups {
+		if sg.ID != nil && sg.Filters != nil {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec.awsLaunchTemplate.AdditionalSecurityGroups"), "either ID or filters should be used"))
```

Comment:
Should we also log a warning if ARN is specified to say its deprecated?

Reply:
Done